R/utils-json.R

```diff
@@ -263,10 +263,15 @@ subsetByMonth <- function(pts, onlyMonth) {
   }
   return(pts)
 }
+
 getYvals_approvals <- function(object, num_vals){
   ylim <- ylim(object)$side.2[1]
   yvals <- rep(ylim, num_vals)
-  return(yvals)
+  
+  # The constant scaling factor below was reverse-engineered from analyzing the 
+  # (pandoc?) SVG in Inkscape. I'm not entirely sure why it works, and it's easy
+  # to imagine why it wouldn't in some cases.
+  return(0.908196 * yvals)
```

Comment:
I would leave that function as-is and add one that calculates the user coords

```r
getYvals_userExtent <- function(object, num_vals){
  #calculate value using 4% thingy
  usr_coords <- par("usr")[3:4]
  
  # ylim <- final value after calculation (taking log and reverse situations into account)
  # for logged 4% = (log(ylim(object)$side.2[2]) - log(ylim(object)$side.2[1]))*0.04
  # and par("usr")[3] = log(ylim(object)$side.2[1]) - 4%

  yvals <- rep(ylim, num_vals)
  return(yvals)
}
```

then, in `getApprovals`, 
```r
y0 = substitute(getYvals_approvals(plot_object, 1))

#and

y1 = substitute(getYvals_userExtent(plot_object, 1))
```